transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -315,7 +315,7 @@ static jstring netty_epoll_native_kernelVersion(JNIEnv* env, jclass clazz) {
 }
 
 static jboolean netty_epoll_native_isSupportingSendmmsg(JNIEnv* env, jclass clazz) {
-    if (sendmmsg) {
+    if (&sendmmsg) {
```

Comment:
There are 2 things at play here:

1. Compile time availability (guarded by `IO_NETTY_SENDMMSG_NOT_FOUND`)
2. Link time availability (guarded by `__attribute__((weak))`)

IIUC your configuration makes it such that `IO_NETTY_SENDMMSG_NOT_FOUND` is not defined and therefore you have the function pointer `sendmmsg` defined at compile time. However this doesn't say anything about what happens at run time (imagine linking this artifact on a pre 3.0 kernel). The compiler warning doesn't doesn't seem to be taking into account the soft linking portion of the method definition. Are you sure that the address for the method pointer will also be `0` at run time? This seems to be the case in the environment I tried (see code below) (gcc 5.4.0, ldd 2.23, Ubuntu 16.10) but I'm only confident that the pointer will point to `NULL`. Is there a reference somewhere you can point me to?


```c
#include <stdlib.h>
#include <stdio.h>

extern int softlinkmethod() __attribute__((weak));

int main() {
    int* f = NULL;
    printf("ref: %p deref: %p\n", f, &f);
    printf("ref: %p deref: %p\n", softlinkmethod, &softlinkmethod);
    if (softlinkmethod != NULL) {
        printf("1 in if\n");
    } else {
        printf("2 in else\n");
    }
    if (&softlinkmethod) {
        printf("3 in if\n");
    } else {
        printf("4 in else\n");
    }
    return 0;
}

/*int softlinkmethod() {
  return 1;
}*/
```

#### stdout
```
ref: (nil) deref: 0x7ffca533ba70
ref: (nil) deref: (nil)
2 in else
4 in else
```